### PR TITLE
[FLOC-4506] Remove the obsolete container API tests

### PR DESCRIPTION
**Pull request by wallrj**
Fixes: https://clusterhq.atlassian.net/browse/FLOC-4506

Also removed container agent related flocker-diagnostics tests and various ancillary code which appears not to be used elsewhere.

This is the same as #2914 but merging into the remove-container-api-FLOC-4505 feature branch instead of into master.

### CODE REVIEW

File: flocker/acceptance/integration/testtools.py [hunk 1]
```diff
@@ -3,11 +3,163 @@
...
 
-from ..testtools import require_cluster, create_dataset
+from ..testtools import require_cluster, create_dataset, get_docker_client
 from ...testtools import AsyncTestCase, random_name
 
 
+DOCKER_CLIENT_LOCK = DeferredLock()
+
+
+def deferToThreadWithLock(lock, f, *args, **kwargs):
```
Comment (wallnerryan) [inline flocker/acceptance/integration/testtools.py]: CamelCase?

File: flocker/acceptance/integration/testtools.py [hunk 2]
```diff
@@ -3,11 +3,163 @@
...
+                                external_port):
+    """
+    Create and start a ``Container`` on ``node``.
+    Clean it up when the ``test`` has completed.
+
+    :param TestCase test: The test.
+    :param Cluster cluster: The ``Cluster`` with certificates for
+        authenticating with the docker daemon on ``node``.
+    :param unicode image_name: A name of the Docker image to use.
+    :param Dataset dataset: The mounted Flocker dataset to bind mount into the
```
Comment (wallnerryan) [inline flocker/acceptance/integration/testtools.py]: How do we cleanup this dataset? I think im missing where that happens.

File: flocker/acceptance/integration/testtools.py (hunk 1 above)
Comment (wallrj) [inline flocker/acceptance/integration/testtools.py]: Not for functions.

File: flocker/acceptance/integration/testtools.py (hunk 2 above)
Comment (wallrj) [inline flocker/acceptance/integration/testtools.py]: The ``create_dataset(test=self, ...)`` function registers a ``TestCase.addCleanup`` function to remove the dataset using a Flocker API call.

**Comment by wallrj** [comment]
Thanks for the review @wallnerryan.
Merging now...

**Pull request merged**